### ISSUE TITLE
[BUG] DP7 Arm64 Console fails due to SDK load problem (includes workaround)

### ISSUE THREAD

**Issue by Psychlist1972** [microsoft/MIDI#430]
I'm still debugging the root cause here. It may impact all Arm64 apps using the SDK. When running the console, if you see a COM Interop exception 0x800401F9 CO_E_ERRORINDLL, then read on.

![Image](https://github.com/user-attachments/assets/eedc5db1-069a-426e-9be7-25e6edaf98df)

To work around the console problem, you can use the Intel/AMD x64 version of the SDK and tools while continuing to use the Arm64 version of the service. This will keep the service native, while running the client tools under emulation.

**Workaround Instructions, for Arm64 devices only**

Uninstall the Arm64 Tools & SDKs install. Leave the Arm64 service install in place, as it will remain Arm64.

Install the Intel/AMD x64 Windows MIDI Services Tools and SDK installer from the release.
https://github.com/microsoft/MIDI/releases/tag/dev-preview-7

Install the Intel/AMD x64 version of the .NET 8 Desktop Runtime from here
https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-8.0.10-windows-x64-installer

In the end, you should see mismatched installs like this:

![Image](https://github.com/user-attachments/assets/3e48e1ce-1bdc-4a14-b60d-912b6ddc8ca8)

At that point, the console and other in-box tools from this release will work under emulation.

**Comment by AmeNote-MikeKent** [comment]
I have found fleeting success in using the workaround as suggested by Pete. Often when running the Console command "midi endpoint monitor" it hangs, never shows any endpoints (not even the loopback endpoints). I have not been able to isolate the difference between when it does work and when it does not. I have spent several hours trying.

Also: I found that if I do not unplug and reconnect after updating the device to use the USBMidi2 driver, then a green screen occurs. I believe this is the prior bug #181, which has been closed. Perhaps it is fixed on x64 but not on arm64? Or an anomaly of the workaround?
